Sources/Caching/PostedTransactionCache.swift

```diff
@@ -32,15 +35,41 @@ final class PostedTransactionCache: PostedTransactionCacheType {
     }
 
     func savePostedTransaction(_ transaction: StoreTransactionType) {
+        RCIntegrationTestAssertNotMainThread()
```

Comment:
Having a large cache can make this unnecessarily slow because we need to load them all in memory. This ensures at least we don't do that in the main thread.